src/views/migrationpolicies/list/components/MigrationPolicyCreateForm/copmonents/SelectorLabelMatchGroup/SelectorLabelMatchGroup.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { Label, LabelGroup } from '@patternfly/react-core';
+
+import { extractKeyValueFromLabel, transformKeyValueToLabel } from './utils';
+
+import './SelectorLabelMatchGroup.scss';
+
+type SelectorLabelMatchGroupProps = {
+  labels: { [key: string]: string };
+  setLabels?: React.Dispatch<
+    React.SetStateAction<{
+      [key: string]: string;
+    }>
+  >;
+  isVMILabel?: boolean;
+};
+
+const SelectorLabelMatchGroup: React.FC<SelectorLabelMatchGroupProps> = ({
+  labels,
+  setLabels,
+  isVMILabel,
+}) => {
+  const { t } = useKubevirtTranslation();
+
+  return (
+    <>
+      <LabelGroup numLabels={10} className="mp-match-label-group" isEditable>
+        {Object.keys(labels || {})?.map((key) => {
+          return (
+            <Label
+              className={classNames({ 'kv-migration-policy__label-vm': isVMILabel })}
+              key={key}
+              color={isVMILabel ? 'grey' : 'blue'}
+              isEditable
+              onClose={() =>
+                setLabels((prevLabels) => {
+                  const updatedLabels = { ...prevLabels };
+                  delete updatedLabels[key];
+                  return updatedLabels;
+                })
+              }
```

Comment:
Would be great to have here something like 

```js
const onDeleteLabel = (key) => () => setLabels((prevLabels) => {
    const updatedLabels = { ...prevLabels };
    delete updatedLabels[key];
    return updatedLabels;
  })
```

and use it in the onClose

```js
onClose={onDeleteLabel(key)}
```

WDYT?

Reply:
I'm not sure I understand what we gain from such a change, can you please explain the advantages of this change please?

Reply:
I guess the name of the function could help for the readabilty of this, will change thanks :)

Reply:
I took it further and extracted all of the anonymous functions out to const to have better readability, Thanks! :)